### PR TITLE
Add support for attributes and delay seconds for SQS

### PR DESCRIPTION
**Pull request by nickcaballero**
Adds support for sending SQS attributes and setting the delay seconds.

### CODE REVIEW

**Comment by viggu888** [comment]
Hi,
I tried the below code on my local and getting this error while sending the message using that method with message attributes,

await sqs.sendMessage(testQueue, JSON.stringify({ value: '123' }), {
      messageAttributes: {
          'my-attribute': {
              type: 'String',
              value: 'my-attribute-value'
          }
      }
  }) 

ERRO[0001] Uncaught (in promise) SQSServiceError: The type of message (user) attribute 'my-attribute' is invalid. You must use only the following supported type prefixes: Binary, Number, String.
running at bound construct (native)

**Comment by nickcaballero** [comment]
> Hi, I tried the below code on my local and getting this error while sending the message using that method with message attributes,
> 
> await sqs.sendMessage(testQueue, JSON.stringify({ value: '123' }), { messageAttributes: { 'my-attribute': { type: 'String', value: 'my-attribute-value' } } })
> 
> ERRO[0001] Uncaught (in promise) SQSServiceError: The type of message (user) attribute 'my-attribute' is invalid. You must use only the following supported type prefixes: Binary, Number, String. running at bound construct (native)

Ah yes, mixed up the value attribute with the `DataType`.  Can you please try again?

File: src/internal/sqs.ts [hunk 1]
```diff
@@ -64,6 +64,22 @@ export class SQSClient extends AWSClient {
             body = { ...body, MessageGroupId: options.messageGroupId }
         }
 
+        if (typeof options.messageAttributes !== 'undefined') {
+            const attributeParameters = Object.entries(options.messageAttributes).reduce((params, [name, attribute], i) => {
```
Comment (oleiade) [inline src/internal/sqs.ts:68]: This specific code would deserve some comment explaining what's going on. I had to read the docs and execute and log the code to figure it out myself: based on the attributes type, we match it to a specific object shape with the value field being different for each 😄 🙇🏻

**Review by oleiade — COMMENTED** [review]
LGTM 👍🏻 🙇🏻 

Left a request for a comment, to make the code easier to maintain, but great job overall and thank you so much for your contribution.

Once the other issue raised is addressed, I'll happily approve and merge this 🙇🏻

File: src/internal/sqs.ts (hunk 1 above)
Comment (nickcaballero) [reply src/internal/sqs.ts:74]: Done!

Comment (viggu888) [reply src/internal/sqs.ts:74]: @nickcaballero It worked now. Thanks for your help.

Comment (oleiade) [reply src/internal/sqs.ts:74]: Merci 🙇🏻

**Pull request merged**